### PR TITLE
feat(ListItem,Button,ToolbarItem,ToolbarMenu): Adding/fixing universal focus indicator

### PR DESCRIPTION
**Pull request by bcalvery**
Updating ListItem, Button, ToolbarItem, ToolbarMenuItem to use the common "getBorderFocusStyles" component to indicate keyboard focus.

Also incorporated theme changes for button - fixing issues reported by design.

**Note:** Keyboard focus is now a distinctly separate treatment from hover.

## ListItem:
Selected item at bottom of each list, keyboard focused item at top.
Before:
image
After:
image

## Button:
Issue with border was in the wrong position (it should be outboard of main button border)
Idle state on left, keyboard focus state on right.
Before:
image
After:
image

Before:
image
After:
image

Note style of circular button is changed to behave like normal button.
Also updated buttonVariables to use colorScheme variables.

## ToolbarItem
Before:
image
After:
image

## ToolbarMenuItem:
Before:
image
After:
image

### CODE REVIEW

**Review by codepretty — APPROVED** [review]
I keyboarded around a few components and looks good from what I can tell! thanks Brian!!

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Fixes
 - Update `call-missed-line` icon in Teams theme @codepretty ([#2059](https://github.com/stardust-ui/react/pull/2059))
 - Show debug panel correctly for components with no owner @miroslavstastny ([#2055](https://github.com/stardust-ui/react/pull/2055))
+- Update `listItem`, `button`, `toolbar`, `toolbarMenu` to use `getBorderFocusStyles` @bcalvery ([#2011](https://github.com/stardust-ui/react/pull/2011))
```
Comment (codepretty) [inline CHANGELOG.md]: I would add "in the Teams theme" here

File: packages/react/src/themes/teams/components/Button/buttonVariables.ts [hunk 2]
```diff
@@ -82,45 +76,39 @@ export default (siteVars: any): ButtonVariables => ({
...
-  primaryBackgroundColor: siteVars.colors.brand[600],
-  primaryBackgroundColorActive: siteVars.colors.brand[900],
-  primaryBackgroundColorHover: siteVars.colors.brand[800],
-  primaryBackgroundColorFocus: siteVars.colors.brand[800],
+  primaryColor: siteVars.colorScheme.brand.foreground4,
+  primaryColorHover: siteVars.colorScheme.brand.foreground4,
+  primaryBackgroundColor: siteVars.colorScheme.brand.background,
+  primaryBackgroundColorActive: siteVars.colorScheme.brand.backgroundPressed,
+  primaryBackgroundColorHover: siteVars.colorScheme.brand.backgroundHover,
+  primaryBackgroundColorFocus: '',
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Button/buttonVariables.ts]: should this be the same as primaryBackgroundColor now? Can we replace the `''` with some value or remove the variable?

File: packages/react/src/themes/teams/components/List/listItemStyles.ts [hunk 3]
```diff
@@ -28,42 +29,44 @@ const selectableHoverStyle = (p: ListItemProps, v): ICSSInJSStyle => ({
...
+      ...((p.selectable || p.navigable) && {
+        position: 'relative',
+
+        // hide the end media by default
+        [`& .${ListItem.slotClassNames.endMedia}`]: { display: 'none' },
+
+        '&:hover': selectableHoverStyle(p, v),
+        ':focus': {
+          ...borderFocusStyles[':focus'],
+        },
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/List/listItemStyles.ts]: `':border': borderFocusStyles[':focus'],`

File: packages/react/src/themes/teams/components/Toolbar/toolbarCustomItemStyles.ts [hunk 4]
```diff
@@ -32,13 +38,11 @@ const toolbarCustomItemStyles: ComponentSlotStylesPrepared<
         }),
 
       ':focus': {
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarCustomItemStyles.ts]: `':focus': borderFocusStyles[':focus'],`

File: packages/react/src/themes/teams/components/Toolbar/toolbarCustomItemStyles.ts [hunk 5]
```diff
@@ -32,13 +38,11 @@ const toolbarCustomItemStyles: ComponentSlotStylesPrepared<
...
-        outline: 0,
+        ...borderFocusStyles[':focus'],
       },
 
       ':focus-visible': {
-        color: v.foregroundFocus || colors.foregroundFocus,
-        backgroundColor: v.backgroundFocus || colors.backgroundFocus,
-        borderColor: v.borderFocus || colors.borderFocus,
+        ...borderFocusStyles[':focus-visible'],
       },
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarCustomItemStyles.ts]: `':focus-visible': borderFocusStyles[':focus-visible'],`

File: packages/react/src/themes/teams/components/Toolbar/toolbarItemStyles.ts [hunk 6]
```diff
@@ -37,10 +43,7 @@ const toolbarItemStyles: ComponentSlotStylesPrepared<ToolbarItemProps, ToolbarVa
       },
 
       ':focus-visible': {
-        color: v.foregroundFocus || colors.foregroundFocus,
-        backgroundColor: v.backgroundFocus || colors.backgroundFocus,
-        borderColor: v.borderFocus || colors.borderFocus,
-        ...getIconFillOrOutlineStyles({ outline: false }),
+        ...borderFocusStyles[':focus-visible'],
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarItemStyles.ts]: https://github.com/microsoft/fluent-ui-react/pull/2011/files#r339978282

File: packages/react/src/themes/teams/components/Toolbar/toolbarItemStyles.ts [hunk 7]
```diff
@@ -21,7 +27,7 @@ const toolbarItemStyles: ComponentSlotStylesPrepared<ToolbarItemProps, ToolbarVa
       cursor: 'pointer',
 
       ':focus': {
-        outline: 0,
+        ...borderFocusStyles[':focus'],
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarItemStyles.ts]: https://github.com/microsoft/fluent-ui-react/pull/2011/files#r339977994

File: packages/react/src/themes/teams/components/Toolbar/toolbarMenuItemStyles.ts [hunk 8]
```diff
@@ -27,8 +34,7 @@ const toolbarMenuItemStyles = {
       },
 
       ':focus-visible': {
-        color: v.menuItemForegroundFocus || colors.menuItemForegroundFocus,
-        backgroundColor: v.menuItemBackgroundFocus || colors.menuItemBackgroundFocus,
+        ...borderFocusStyles[':focus-visible'],
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarMenuItemStyles.ts]: https://github.com/microsoft/fluent-ui-react/pull/2011/files#r339978282

File: packages/react/src/themes/teams/components/Toolbar/toolbarVariables.ts [hunk 9]
```diff
@@ -86,7 +78,6 @@ export default (siteVars: any): ToolbarVariables => ({
         borderFocus: siteVars.colorScheme.brand.borderFocus1,
         foregroundHover: siteVars.colorScheme.brand.foregroundHover,
         backgroundHover: 'transparent',
-        foregroundFocus: siteVars.colorScheme.brand.foregroundFocus,
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Toolbar/toolbarVariables.ts:89]: Why are you removing only this override? Shouldn't the `backgroundFocus` be removed too?

**Pull request merged**